liota/dccs/iotcv2.py

```diff
@@ -0,0 +1,346 @@
+            raise TypeError()
+
+        reg_entity_child.parent = reg_entity_parent
+        if isinstance(reg_entity_child, RegisteredMetric):
+            # should save parent's reg_entity_id
+            reg_entity_child.reg_entity_id = reg_entity_parent.reg_entity_id
+            #entity_obj = reg_entity_child.ref_entity
+            #self.publish_unit(reg_entity_child, entity_obj.name, entity_obj.unit)
+            return
+        else:
+            log.info("Registering parent and child relationship with IoTCV2 for {0} {1}".\
+                     format(reg_entity_parent.ref_entity.name, reg_entity_child.ref_entity.name))
+            # create relationship fun internal queue
+            rel_resp_q = Queue.Queue()
+            self.set_relationship = False
+
+            def on_response(msg, rel_resp_q):
+                try:
+                    log.debug("Received msg: {0}".format(msg))
+                    json_msg = json.loads(msg)
+                    log.debug("Processed msg: {0}".format(json_msg["type"]))
+                    if json_msg["type"] == "createRelationship_rsp" and json_msg["guid"] != "null" \
+                        and json_msg["guid"] == reg_entity_parent.reg_entity_id:
+                        log.info("Received createRelationship response")
+                        if json_msg["result"] == 'SUCCESS':
+                            self.set_relationship = True
+                    else:
+                        log.info("Waiting for create relationship response")
+                        on_response(rel_resp_q.get(True, timeout), rel_resp_q)
+                except:
+                    raise Exception("Exception while create relationship")
+
+            # create and add register request into req_list (waiting list)
+            operation_id = self.next_id()
+            req = Request(reg_entity_parent.reg_entity_id, operation_id, rel_resp_q)
+            log.info("operation_id:{0} req:{1}".format(operation_id, req))
+            with self.req_ops_lock:
+                self.req_dict.update({str(operation_id): req})
+                log.info("operation_id:{0} req_dict:{1}".format(operation_id, self.req_dict))
+            self.comms.send(json.dumps(self._relationship(operation_id,
+                            reg_entity_parent.reg_entity_id, reg_entity_child.reg_entity_id)))
+            # block until there is an item available or timeout
+            try:
+                on_response(rel_resp_q.get(True, timeout), rel_resp_q)
+            except:
+                raise Exception("Exception while creating relationship")
+
+            if not self.set_relationship:
+                raise CreateRelationshipFailure()
+            log.info("Relationship Created")
+
+    def _format_data(self, reg_metric):
+        met_cnt = reg_metric.values.qsize()
+        if met_cnt == 0:
+            return
+        data = []
+        for _ in range(met_cnt):
+            m = reg_metric.values.get(block=True)
```

Comment:
Do we need to specify block=True? Isn't it True by default?

Reply:
isn't better to set it explicitly like what we have done for IoTCC?